File: CodeChef/November-long-2020/unusual-queries/UNSQUERS.py
num, que, satis = map(int, input().split())
last = 0
a = list(map(int, input().split()))
for _ in range(que):
    x, y = map(int, input().split())
    L = (x+(satis*last)-1) % num+1
    R = (y+(satis*last)-1) % num+1
    if(L > R):
        L, R = R, L
    for i in range(num):
        for j in range(i+1, num):
            q = 0
            m = []
            for k in range(i, j-1):
                ans = 1
                temp = a[k]
                for l in range(k+1, j-1):
                    if a[l] > temp:
                        temp = a[l]
                        ans += 1
                m.insert(q, ans)
                q += 1
            m.sort()
    temp = ans = l = 0
    b = []
    for y in range(L-1, R):
        ans = 1
        temp = a[y]
        for j in range(y+1, R):
            if a[j] > temp:
                temp = a[j]
                ans += 1
        b.insert(l, ans)
        l += 1
    b.sort()
    print(b[l-1])
    last = b[l-1]

File: CodeChef/November-long-2020/unusual-queries/UNSQUERS.cpp
#include <bits/stdc++.h>
using namespace std;

int main()
{
    int num, que, satis, L, R, x, y, last = 0, z;
    cin >> num >> que >> satis;
    int a[num];
    for (int i = 0; i < num; i++)
    {
        cin >> a[i];
    }

    for (int u = 0; u < que; u++)
    {
        cin >> x;
        cin >> y;

        L = (x + (satis * last) - 1) % num + 1;
        R = (y + (satis * last) - 1) % num + 1;

        if (L > R)
        {
            swap(L, R);
        }
        for (int i = 0; i < num; i++)
        {
            for (int j = i + 1; j < num; j++)
            {
                int q = 0, m[num] = {'\0'};
                for (int k = i; k < j - 1; k++)
                {
                    int ans = 1;
                    int temp = a[k];
                    for (int l = k + 1; l < j - 1; l++)
                    {
                        if (a[l] > temp)
                        {
                            temp = a[l];
                            ans++;
                        }
                    }
                    m[q++] = ans;
                }
                sort(m, m + q);
            }
        }
        int temp = 0, ans = 0;
        int b[R - L + 1] = {'\0'}, l = 0;
        for (int y = L - 1; y < R; y++)
        {
            ans = 1;
            temp = a[y];
            for (int j = y + 1; j < R; j++)
            {
                if (a[j] > temp)
                {
                    temp = a[j];
                    ans++;
                }
            }
            b[l++] = ans;
        }
        sort(b, b + l);
        cout << b[l - 1] << endl;
        last = b[l - 1];
    }

    return 0;
}